test/jest/__tests__/utils/utils.DependencyUtils.ts.spec.ts

```diff
@@ -0,0 +1,280 @@
+    });
+
+    modPack = createCombo({
+        name: "modPack",
+        versionNumber: "1.0.0",
+        dependencies: ["author-mod-1.0.0"],
+        categories: ["Modpacks"]
+    });
+
+    modWithLoopDependencyA = createCombo({
+        name: "modWithLoopDependencyA",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithLoopDependencyB-1.0.0"]
+    });
+
+    modWithLoopDependencyB = createCombo({
+        name: "modWithLoopDependencyB",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithLoopDependencyA-1.0.0"]
+    });
+};
+
+describe("DependencyUtils.getFullDependencyList", () => {
+
+    beforeEach(() => {
+        beforeSetup();
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe("Empty combo list", () => {
+        it("should return an empty list", async () => {
+            const combos: ThunderstoreCombo[] = [];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, []);
+        });
+    });
+
+    describe("Single mod without dependencies", () => {
+        it("should return single mod unchanged", async () => {
+            const combos = [mod];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, ["author-mod-1.0.0"]);
+        });
+    });
+
+
+    describe("Single mod with a dependency", () => {
+        it("should return the mod and its dependency's latest version", async () => {
+            const combos = [modWithDependency];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-mod-2.0.0",
+                "author-modWithDependency-1.0.0"
+            ]);
+        });
+    });
+
+
+    describe("Two mods both having a dependency to the same mod", () => {
+        it("should return the two mods at their specific versions and the dependency's latest version", async () => {
+            const combos = [modWithDependency, modWithTheSameDependency];
+
+            await getFullDependencyList(combos, game, []);
+            
+            expectToContainOnly(combos, [
+                "author-mod-2.0.0",
+                "author-modWithDependency-1.0.0",
+                "author-modWithTheSameDependency-1.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A mod with a deep dependency", () => {
+        it("should return the mod and its dependency's last version and its dependency's dependency's latest version", async () => {
+            const combos = [modWithDeepDependency];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-modWithDeepDependency-1.0.0",
+                "author-modWithDependency-2.0.0",
+                "author-mod-2.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A mod with a loop dependency", () => {
```

Comment:
- What happens if A1.0.0 Depends on B1.0.0, but B2.0.0 depends on A2.0.0?
- What happens in the currently tested scenario, or the one above, if A is a modpack?

Reply:
These are now demonstrated in test cases:
- What happens if A1.0.0 Depends on B1.0.0, but B2.0.0 depends on A2.0.0?
  - It will always use the latest version for the dependency

- What happens in the currently tested scenario, or the one above, if A is a modpack?
  - If we start from the mod pack, it uses the **specific** version of the dependency (mod)
  - If we start from the mod, it uses the **latest** version of the dependency (mod pack)